lib/rules/no-noop-setup-on-error-in-before.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+const { getImportIdentifier } = require('../utils/import');
+const types = require('../utils/types');
+
+//------------------------------------------------------------------------------
+// General rule - Disallows noop `setupOnError` in `before` or `beforeEach`.
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallows using noop setupOnerror in `before` or `beforeEach`',
+      category: 'Testing',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-noop-setup-on-error-in-before.md',
+    },
+    fixable: 'code',
```

Comment:
I'm a bit hesitant to make this rule autofixable by simply deleting the offending code. That changes the behavior of the code (which we normally try to avoid) and the author would likely need to make manual fixes anyway right? What do you think?

Reply:
There are certainly more actions that the author would have to do to fix the tests but it would vary on a case by case basis. The one thing that would have to be done anyway - removal of the offending code, then run the tests and see which test cases failed to add error handling in individual `test`. I would leave it for convenience.

Reply:
@mongoose700 thoughts?